### PR TITLE
[SPARK-33111][ML] aft transform optimization

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
1, when `predictionCol` and `quantilesCol` are both set, we only need one prediction for each row: prediction is just the variable `lambda` in `predictQuantiles`;
2, in the computation of variable `quantiles` in `predictQuantiles`, a pre-computed vector `val baseQuantiles = $(quantileProbabilities).map(q => math.exp(math.log(-math.log1p(-q)) * scale))` can be reused for each row;

### Why are the changes needed?
avoid redundant computation in transform, like what we did in `ProbabilisticClassificationModel`, `GaussianMixtureModel`, etc

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
existing testsuite

### CODE REVIEW

**Comment by zhengruifeng** [comment]
friendly ping @srowen @huaxingao

File: mllib/src/main/scala/org/apache/spark/ml/regression/AFTSurvivalRegression.scala [hunk 1]
```diff
@@ -421,9 +421,17 @@ class AFTSurvivalRegressionModel private[ml] (
     }
 
     if (hasQuantilesCol) {
-      val predictQuantilesUDF = udf { features: Vector => predictQuantiles(features)}
+      val baseQuantiles = $(quantileProbabilities)
+        .map(q => math.exp(math.log(-math.log1p(-q)) * scale))
+      val lambdaCol = if ($(predictionCol).nonEmpty) {
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/regression/AFTSurvivalRegression.scala:426]: I think you're probably right here but why is lambdaCol the first predictionCol here?

Comment (zhengruifeng) [reply mllib/src/main/scala/org/apache/spark/ml/regression/AFTSurvivalRegression.scala:426]: if ($(predictionCol).nonEmpty), then `predictionColumns.head` is `predictUDF(col($(featuresCol))).as($(predictionCol), outputSchema($(predictionCol)).metadata)`, the predictionCol.

**Review by huaxingao — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Comment by zhengruifeng** [comment]
Thank you for reviewing!

**Pull request closed without merging**